### PR TITLE
add skaffold support for fluent-bit-plugin

### PR DESCRIPTION
**Pull request by nickytd**
/area logging
/kind test

This PR brings `skaffold` development scenario for the `fluent-bit-vali-plugin`. It builds upon the gardener skaffold pipeline. 

```other developer
Introduces a skaffold local development pipeline to fluent-bit-vali-plugin
```

### CODE REVIEW

File: docs/development/skaffold.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+It tries to get the version of which gardener release to fetch from the dependency defined in the project `go.mod` file.
+Once the gardener repository is fetched, initiate the kind cluster with:
+```bash
+make -C gardener kind-up
+```
+
+In addition to the kind cluster itself, the `kind-up` target brings also: a docker registry, calico and a metrics-server.
+Once the kind cluster is up and running we can start the `skaffold` local pipeline with the following command:
+```bash
+make skaffold-up
```
Comment (vlvasilev) [inline docs/development/skaffold.md:25]:
```suggestion
make skaffold-run
```

File: docs/development/skaffold.md [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+```
+
+This uses the skaffold.yaml definition to bring `etcd`, `controlplane`,  `provider local` and `gardenlet` skaffold modules into the local kind cluster. The last `gardenlet` module contains the build target for generating the `fluent-bit-plugin` container image and it pushes it to the local registry already present in the local kind cluster. Once the skaffold deployment is triggered the `fluent-bit-plugin` is brought into the cluster with the gardenlet specific deployment model.
+
+After the `skaffold-up` target is executed the skaffold pipeline may be started in the `dev` mode with
```
Comment (vlvasilev) [inline docs/development/skaffold.md:30]: `skaffold-run` is in the Makefile

Also, there are inconsistencies between `skaffold` commands in the Makefile and those on the picture.
Does the `skaffold-run` correspondence to the `skafold build` and `scaffold-dev` to `skaffold deploy`?

**Review by vlvasilev — CHANGES_REQUESTED** [review]

File: docs/development/skaffold.md (hunk 1 above)
Comment (nickytd) [reply docs/development/skaffold.md]: Thanks the feedback is applied.

File: docs/development/skaffold.md (hunk 2 above)
Comment (nickytd) [reply docs/development/skaffold.md]: Thanks I updated the image diagram to outline that it depicts skaffold workflow phases. And improved the documentation to make the missing link from make targets and skaffold workflow phases.

**Pull request merged**